### DISCUSSION TITLE
whether casks should be replaced directly during the upgrade

### DISCUSSION THREAD

**Discussion by Misaka13514** [Homebrew/discussions#1205] [Casks] [answered]
**My launchpad becomes a mess** after the upgrade of casks. 🤔 
If they are directly replaced, instead of deleting and adding, that won't happen.

But that may cause conflicts or errors(?)

Any opinion?

**Comment by SMillerDev** [comment]
They can't always be replaced, and this causes it's own set of issues. How would that work for casks based on a `.pkg` file for example?

**Reply by Misaka13514** [reply]
So... we need to judge

**Comment by carlocab** [comment, marked as answer]
I think there was an issue at Homebrew/cask regarding this. It might be closed now, as I didn't find it from scanning the open issues (but maybe I just didn't look hard enough).

From what I recall, this is certainly something that would be nice to have but is actually tricky to implement. However, if you know how to do this, please do open a pull request.

**Reply by SMillerDev** [reply]
I think you're refering to https://github.com/Homebrew/homebrew-cask/issues/102721

**Reply by carlocab** [reply]
So I just didn't look hard enough. Thanks for outing me 😅
